Chapter7/db_revisit.erl

```diff
@@ -5,15 +5,15 @@
 % Exercise 7-3: The db.erl Exercise Revisited
 % Don't know what the book means by "Test your results using the database server developed in Exercise 5-1 in Chapter 5."
 
-new() 				-> [].
-destroy(_) 			-> ok.
-write(Data, Db) 	-> [Data | Db].
-delete(Data, Db) 	-> Db -- match(Data, Db).
+new()             -> [].
+destroy(_)        -> ok.
+write(Db, Data)   -> [Data | Db].
+delete(Db, Data)  -> Db -- match(Db, Data).
 
-read(_, [])										-> {error, key_not_found};
-read(#data{key=Key}, [New = #data{key=Key}|_])	-> {ok, New#data.data};
-read(#data{key=Key}, [_|Tail])					-> read( #data{key=Key}, Tail ).
+read(_, [])                                           -> {error, key_not_found};
```

Comment:
I might be missing something. This is not an error, but just a design question.
Why do you use `#data{}` to find a `data` element by key?
Wouldn't it be better if you had used `Key` as a first element instead? You don't need the whole constructed `#data{}` to find by key, do you?
The same applies for `match`  function.

Reply:
Not sure what your asking, if its about me using "#data{key=Key, data=Data}" its because thats how the book recommends doing it:

> You could define the function using field selection and update like this:
birthday(P) ->
P#person{age = P#person.age + 1}.
But it is clearer to use pattern matching:
birthday(#person{age=Age} = P) -> P#person{age=Age+1}.

>